src/yaml/yamlUtils.ts

```diff
@@ -308,3 +311,135 @@ export function convertJsonSnippetToYaml(snippetText: string) {
         })
         .join('\n')
 }
+
+function getNumberOfLeftSpaces(text: string) {
+    let numOfLeftSpaces = 0
+
+    for (const char of text) {
+        if (char === ' ') {
+            numOfLeftSpaces++
+        } else {
+            break
+        }
+    }
+
+    return numOfLeftSpaces
+}
+
+// tslint:disable:cyclomatic-complexity
+export function getOffsetData(document: TextDocument, offset: number) {
```

Comment:
this function looks a bit complicated, is there a way break it into smaller ones?

Reply:
I broke it down to 2 functions - one iterates backwards and the other forwards. I also removed some extraneous code. I noticed that I was splitting the whole text into an array which was defeating the purpose of using for loop - better performance.